lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1724,4 +1726,138 @@ public function validate_user_can_query_private_statuses( $value, $request, $par
 		return new WP_Error( 'rest_forbidden_status', __( 'Status is forbidden' ), array( 'status' => rest_authorization_required_code() ) );
 	}
 
+	/**
+	 * Validate filter param.
+	 *
+	 * This function can be used to validate query params passed by the filter request param.
+	 *
+	 * @param  mixed           $value     Value of the parameter.
+	 * @param  WP_REST_Request $request   Request body.
+	 * @param  string          $parameter Name of the parameter. Should be tax_query.
+	 * @return WP_Error|boolean
+	 */
+	public function rest_validate_query_filter( $values, $request, $parameter ) {
+		foreach ( $values as $parameter => $value ) {
+			switch ( $parameter ) {
+				case 'tax_query' :
+					$is_valid = $this->rest_validate_tax_query( $value, $request, $parameter );
+					// If an error is found exit early and it will throw an error for request->has_valid_params();
+					if ( is_wp_error( $is_valid ) ) {
+						return $is_valid;
+					}
+					continue;
+				default :
+					// Returns true for validation by default.
+					$is_valid = true;
+					continue;
+			}
+		}
+		return $is_valid;
```

Comment:
If there are no `$values` then would't this code throw an error since `$is_valid` is not declared anywhere?  I would just `return true` here if you want that to be the default value, or declare `$is_valid = true` before the foreach.